includes/rest/class-url-validator-controller.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * ActivityPub URL Validator Controller.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use WP_REST_Controller;
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use Activitypub\Http;
+
+use function Activitypub\get_embed_html;
+use function Activitypub\is_activity;
+
+/**
+ * URL Validator Controller Class.
+ */
+class URL_Validator_Controller extends WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'url/validate';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'validate' ),
+					'args'                => array(
+						'url' => array(
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'esc_url_raw',
```

Comment:
```suggestion
							'format' => 'uri',
							'required' => true,
```
With these changes most checks in `validate()` should be possible to remove

Reply:
Excellent. Nice to have a REST API sleuth on the job to compensate for my Windsurf slop ;)